src/main/java/de/tum/in/www1/artemis/service/SubmissionService.java

```diff
@@ -531,4 +576,54 @@ public void checkIfExerciseDueDateIsReached(Exercise exercise) {
         return submissions;
     }
 
+    /**
+     * This method gets all complaints of an exercise and returns them together with their corresponding submission in a DTO
+     *
+     * @param exerciseId the exerciseId of the exercise of which the complaints are fetched
+     * @param principal the current user
+     * @param isAtLeastInstructor if the user is an instructor
+     * @return a list of DTOs containing a complaint and its submission
+     */
+    public List<SubmissionWithComplaintDTO> getSubmissionsWithComplaintsForExercise(Long exerciseId, Principal principal, boolean isAtLeastInstructor) {
+        List<SubmissionWithComplaintDTO> submissionWithComplaintDTOs = new ArrayList<>();
+
+        // get all complaints which belong to the exercise
+        List<Complaint> complaints = complaintRepository.getAllComplaintsByExerciseIdAndComplaintType(exerciseId, ComplaintType.COMPLAINT);
+        var complaintMap = complaints.stream().collect(Collectors.toMap(complaint -> complaint.getResult().getId(), value -> value));
+
+        if (complaints.isEmpty()) {
+            return submissionWithComplaintDTOs;
+        }
+        // get the ids of all results which have a complaint, and with those fetch all their submissions
+        List<Long> submissionIds = complaints.stream().map(complaint -> complaint.getResult().getSubmission().getId()).collect(toList());
+        List<Submission> submissions = submissionRepository.findBySubmissionIdsWithEagerResults(submissionIds);
+
+        // add each submission with its complaint to the DTO
+        submissions.forEach(submission -> {
+            Result resultWithComplaint = submission.getResultWithComplaint();
+            if (resultWithComplaint == null) {
+                return;
```

Comment:
The `return` here acts as a `continue`, since it is in a forEach loop. Inside of it you can't use `continue` for some reason. 
For each (submission) the inner function is called, and this inner function returns if `resultWithComplaint` is null. (that's also why the return type is void, unlike the return type of the actual method)